### ISSUE TITLE
Configurable keybindings

### ISSUE THREAD

**Issue by vidosits** [jesseduffield/lazydocker#48]
**Is your feature request related to a problem? Please describe.**
Default keybindings are hard coded. Problem is, not every keyboard has `[` and `]` keys.

**Describe the solution you'd like**
Make the keybindings configurable.

**Describe alternatives you've considered**
Move the `previous/next tab` keybindings from `[` and `]` to something else.

**Comment by glvr182** [comment]
A way to solve this is to use the config file and [this little library](https://github.com/awesome-gocui/keybinding)

**Comment by jesseduffield** [comment]
very cool library! I wonder how best to go about this: we currently have most (though not all) of our keybindings defined in one place, so for those which are stored there, it would not be hard to merge in what we have defined in the user config. In fact even in the other places it wouldn't be hard just to refer to the user config.

in config.yml we can have:
```
keybindings:
  stopContainer: 's'
  nextTab: '['
  previousTab: ']'
```

then in keybindings.go we can have:
```
		{
			ViewName:    "containers",
			Key:         keybinding.MustParse(gui.Config.UserConfig.Keybindings.StopContainer),
			Modifier:    gocui.ModNone,
			Handler:     gui.handleContainerStop,
			Description: gui.Tr.Stop,
		},
```

How does that approach sound? Only real downside with that approach is that you don't get to specify if you want to use the gocui.ModAlt modifier, but we aren't currently using that anywhere anyway.

**Comment by glvr182** [comment]
That approach sounds good, we could indeed look into the modAlt, but that is an issue for the library i guess. I could spend this weekend on it?

**Comment by jesseduffield** [comment]
I agree @glvr182 , and if you want to spend time on this that would be awesome :)

**Comment by jesseduffield** [comment]
Also worth noting that if we set a key to be the empty string, that will enable the user to disable certain keys, which satisfies https://github.com/jesseduffield/lazydocker/issues/27

**Comment by jesseduffield** [comment]
also worth noting we'd need to support arrays of keys which luckily is already covered by that library

**Comment by glvr182** [comment]
@jesseduffield Do you think we should use a map or a struct for the config?

**Comment by jesseduffield** [comment]
I think a struct works better, that way we have strong typing. I made the decision to switch the i18n stuff to a struct and it's already proven really helpful in telling me when things are out of sync :)

**Comment by ser-drephs** [comment]
Any news on this one?

**Comment by jesseduffield** [comment]
This is totally do-able: we can just copy the approach that lazygit uses: https://github.com/jesseduffield/lazygit/blob/master/pkg/config/user_config.go

Any takers?

**Comment by jesperronn** [comment]
@jesseduffield I can see that @dabstractor created a PR. I think the approach as far as my knowledge, follows the direction you mentioned (copy from lazygit).

disclaimer: I have not tried the approach or anything, but reach out if you need help reviewing/testing or similar

**Comment by coolCucumber-cat** [comment]
It's been 6 years. Not to sound impatient but this is taking pretty long for something that is just copy and paste. The default keybindings are absolutely atrocious, especially when you realise that keys are in different places in different countries.

**Comment by dabstractor** [comment]
Yes I followed the advice in this thread to implement it. Working well for me. This repo just seems to be kind of low priority for the author.

**Comment by dabstractor** [comment]
There have been 13 PRs merged in the last year, and 3 closed. At this rate, the PR backlog will be cleared sometime within the next 5 years, assuming no new PRs are submitted in that time. @jesseduffield please delegate what you can't keep up with. If you've moved on and don't want to maintain this repo anymore, maybe a community fork would be best.
